script.js

```diff
@@ -1,4 +1,235 @@
+    return `You entered ${input} which is not a valid input. Only the following are accepted: scissors, paper or stone. ${userName}, please try again.`;
+  }
+
+  var randNumber = randomNumGenerator();
+  var randNumber2 = randomNumGenerator();
+  console.log(randNumber);
+  var handFromProgram = playByProgram(randNumber);
+  var handFromProgram2 = playByProgram(randNumber2);
+
+  if (gameMode == "normal play") {
+    var result = winningHand(input, handFromProgram);
+  }
+  if (gameMode == "reversed play") {
+    var result = reversedGameWinningHand(input, handFromProgram);
+  }
+  if (gameMode == "korean SPS") {
+    myOutputValue = evaluateKoreanSPS(input, handFromProgram);
+    return `You are playing the Korean SPS version. <br><br> You played ${input} while the computer played ${handFromProgram}. <br><br> ${myOutputValue}`;
+  }
+  if (gameMode == "computer vs computer") {
+    input = handFromProgram2;
+    var result = winningHand(handFromProgram, handFromProgram2);
+  }
+  console.log(userWonCount);
+  console.log(comWonCount);
+
+  var winPercent = (
+    (userWonCount * 100) /
+    (userWonCount + comWonCount)
+  ).toFixed(2);
+
+  //if the first round is a draw, since divide by 0 is NaN
+  if (userWonCount == 0 && comWonCount == 0) {
+    winPercent = 0;
+  }
+  myOutputValue = `Hi ${userName}! You played ${input} while the computer played ${handFromProgram}. <br><br> ${result} <br><br> ${userName}'s winning percentage is ${winPercent}%. <br><br> Let's play again!!!`;
   return myOutputValue;
 };
+
+//to process the game mode
+var gameModeProcess = function (userInput) {
+  if (userInput == 1) {
+    return `normal play`;
+  }
+  if (userInput == 2) {
+    return `reversed play`;
+  }
+  if (userInput == 3) {
+    return `korean SPS`;
+  }
+  if (userInput == 4) {
+    return `computer vs computer`;
+  }
+};
+
+//to validate the user input for the game mode
+var modeValidation = function (userInput) {
+  if (
+    !(
+      userInput == "1" ||
+      userInput == "2" ||
+      userInput == "3" ||
+      userInput == "4"
+    )
+  ) {
+    return true;
+  }
+};
+
+//to validate user input
+var inputValidation = function (userGuess) {
+  if (
+    !(userGuess == "scissors" || userGuess == "stone" || userGuess == "paper")
+  ) {
+    return true;
+  }
+};
+
+//to evaluate the winner for normal play
+var winningHand = function (userInput, programPlay) {
+  if (userInput == programPlay) {
+    return "It is a draw!";
+  }
+  if (
+    (userInput == "scissors" && programPlay == "paper") ||
+    (userInput == "paper" && programPlay == "stone") ||
+    (userInput == "stone" && programPlay == "scissors")
+  ) {
+    userWonCount++;
+    return "Congrats, you win!";
+  }
+  if (
+    (userInput == "stone" && programPlay == "paper") ||
+    (userInput == "scissors" && programPlay == "stone") ||
+    (userInput == "paper" && programPlay == "scissors")
+  ) {
+    comWonCount++;
+    return "Oh no, you lost!";
+  }
+};
```

Comment:
great used of function to extract the game logic which make your main function easier to read and awesome use of || 👍